oeps/architectural-decisions/oep-0065-arch-frontend-composability.rst

```diff
@@ -381,6 +381,27 @@ Secondary concerns include:
 Rejected Alternatives
 *********************
 
+Vite Module Federation
+======================
+
+From a modernization and build performance perspective, `Vite <https://vitejs.dev/>`_ is an attractive replacement for Webpack.  It is significantly faster to compile code than Webpack, and is generally more "user friendly" with a slightly higher-level API and configuration.  It takes less configuration code to work with Vite, and it handles more file types by default (such as CSS/SASS, for instance).
+
+Vite has two notable implementations of module federation.  `@originjs/vite-plugin-federation <https://github.com/originjs/vite-plugin-federation>`_ and `module-federation/vite <https://github.com/module-federation/vite>`_.
+
+Unfortunately, both of these implementations have limitations and drawbacks that make them unsuitable as a way of implementing runtime module loading for Open edX:
+
+- We consider runtime remote discovery to be a required feature of our module federation implementation.  This is the ability to register remotes - servers which host modules - at runtime, not just at build time.  We do not want to rebuild our "hosts" - whether shell or another MFE - when we want to add a new remote.  We want to be able to get our remote configuration at runtime and register remotes with the system when the application is running.
```

Comment:
This is the first mention of the term "remote discovery" in the document. I think it'd be nice to have the description of this requirement (including using the term) earlier in the document (possibly in the "Capability: Runtime Module Loading" section).